torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -5841,15 +5844,83 @@ def test_broadcast_object_list(self):
             single_obj_list = [objects[0]]
             if self.rank != src_rank:
                 self.assertNotEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
-            dist.broadcast_object_list(single_obj_list, src=0)
+            dist.broadcast_object_list(single_obj_list, src=src_rank)
             self.assertEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
 
             # Multiple input objects test
             if self.rank != src_rank:
                 self.assertNotEqual(objects, COLLECTIVES_OBJECT_TEST_LIST)
-            dist.broadcast_object_list(objects, src=0)
+            dist.broadcast_object_list(objects, src=src_rank)
             self.assertEqual(objects, COLLECTIVES_OBJECT_TEST_LIST)
 
+        @require_backend({"nccl", "gloo"})
+        @require_n_gpus_for_nccl_backend(
+            int(os.environ["WORLD_SIZE"]), os.environ["BACKEND"]
+        )
+        def test_broadcast_object_list_map_location(self):
+            # Test the `map_location` argument
+            backend = os.environ["BACKEND"]
+            if backend == "nccl":
+                torch.cuda.set_device(self.rank)
+
+            def _check_tensor_map_location(obj, device):
+                # Checks that all tensors contained in `obj` are on `device`
+                # Does not account for tensors stored as fields of a class
+                if isinstance(obj, torch.Tensor):
+                    return obj.device == device
+                if isinstance(obj, (list, tuple)):
+                    elems_match = [_check_tensor_map_location(e, device) for e in obj]
+                    return all(elems_match)
+                if isinstance(obj, collections.abc.Mapping):
+                    values_match = [_check_tensor_map_location(v, device) for _, v in obj.items()]
+                    return all(values_match)
+                return True
```

Comment:
nit: return False because we got some data structure we don't know about? The base case for this recursion should always come down to tensor right? 

This method might return True for an input like [1, 1, 1] right?

Reply:
Great catch. It was legacy from when I was trying to get the function to work with arbitrary classes but failed. I have changed it to `return False` in the base case.